### PR TITLE
feat: Adding support for lock

### LINKED ISSUE #73: 希望在builder添加for update支持
mysql中加锁的语句为``` SELECT * FROM `test` WHERE `id` = 1 FOR UPDATE ```

目前gendry中的builder中没有对`FOR UPDATE`进行支持，只能在构造好的sql后面拼接，如下
```code
cond, vals, _ := builder.BuildSelect(tableName, where, nil)
cond = cond + " for update"
db.Query(cond, vals...)
```

希望可以有单独的方法可以直接构造for update语句，示例（如有其它更好方式也可）：

```code
cond, vals, _ := builder.BuildForUpdateSelect(tableName, where, nil)
db.Query(cond, vals...)
```

### PR DESCRIPTION
**Pull request by sybil-chen**
Closes #73

### CODE REVIEW

**Comment by Colstuwjx** [comment]
LGTM

File: builder/builder.go [hunk 1]
```diff
@@ -19,6 +19,8 @@ var (
 	errLimitValueLength          = errors.New(`[builder] the value of "_limit" must contain one or two uint elements`)
 	errHavingValueType           = errors.New(`[builder] the value of "_having" must be of map[string]interface{}`)
 	errHavingUnsupportedOperator = errors.New(`[builder] "_having" contains unsupported operator`)
+	errLockValueType             = errors.New(`[builder] the value of "_lock" must be of string type`)
+	errNotAllowedLockType        = errors.New(`[builder] the value of "_lock" is not allowed`)
```
Comment (Colstuwjx) [inline builder/builder.go:23]: `_lockMode` as we actually use this word.

Comment (sybil-chen) [reply builder/builder.go:23]: ok, I will be change the error content, thank u.

Comment (sybil-chen) [reply builder/builder.go]: > `_lockMode` as we actually use this word.

what about this?

Comment (Colstuwjx) [reply builder/builder.go]: Sounds good to me, ping @caibirdme for further review.

Comment (sybil-chen) [reply builder/builder.go]: > Sounds good to me, ping @caibirdme for further review.

All right, thank u.

Comment (sybil-chen) [reply builder/builder.go]: @caibirdme Please review the code, thank u.

**Comment by sybil-chen** [comment]
> Good! Could you please add some related documents for this feature?

ok, wait, I will be done it later.

**Comment by sybil-chen** [comment]
@caibirdme Documentation description has been added.

**Comment by caibirdme** [comment]
Really appreciate your work, thx for your contribution!

**Pull request merged**